scripts/o3de/o3de/engine_template.py

```diff
@@ -1997,47 +1990,48 @@ def create_gem(gem_path: pathlib.Path,
+            if not validation.valid_o3de_gem_json(gem_json):
+                logger.error(f'Gem json {gem_json} is not valid.')
+                return 1
+
+            with open(gem_json, 'r') as s:
                 try:
-                    restricted_name = restricted_json_data["restricted_name"]
-                except KeyError as e:
-                    logger.error(f'Failed to read "restricted_name" from restricted json {restricted_json}.')
+                    gem_json_data = json.load(s)
+                except json.JSONDecodeError as e:
+                    logger.error(f'Failed to load gem json {gem_json}.')
                     return 1
 
-                # set the "restricted_name": "restricted_name" element of the gem.json
-                gem_json = gem_path / 'gem.json'
-                if not validation.valid_o3de_gem_json(gem_json):
-                    logger.error(f'Gem json {gem_json} is not valid.')
+            gem_json_data.update({"restricted": restricted_name})
+            os.unlink(gem_json)
+            with open(gem_json, 'w') as s:
+                try:
+                    s.write(json.dumps(gem_json_data, indent=4) + '\n')
+                except OSError as e:
+                    logger.error(f'Failed to write project json {gem_json}.')
                     return 1
 
-                with open(gem_json, 'r') as s:
-                    try:
-                        gem_json_data = json.load(s)
-                    except json.JSONDecodeError as e:
-                        logger.error(f'Failed to load gem json {gem_json}.')
-                        return 1
-
-                gem_json_data.update({"restricted_name": restricted_name})
-                os.unlink(gem_json)
-                with open(gem_json, 'w') as s:
-                    try:
-                        s.write(json.dumps(gem_json_data, indent=4) + '\n')
-                    except OSError as e:
-                        logger.error(f'Failed to write project json {gem_json}.')
-                        return 1
+            for restricted_platform in restricted_platforms:
+                restricted_gem = gem_restricted_path / restricted_platform/ gem_name
+                os.makedirs(restricted_gem, exist_ok=True)
+                cmakelists_file_name = restricted_gem / 'CMakeLists.txt'
+                if not os.path.isfile(cmakelists_file_name):
+                    with open(cmakelists_file_name, 'w') as d:
+                        if keep_license_text:
+                            d.write('# {BEGIN_LICENSE}\n')
+                            d.write('# Copyright (c) Contributors to the Open 3D Engine Project.\n')
+                            d.write('# For complete copyright and license terms please see the LICENSE at the root of this distribution.\n')
+                            d.write('#\n')
+                            d.write('# SPDX-License-Identifier: Apache-2.0 OR MIT\n')
+                            d.write('# {END_LICENSE}\n')
```

Comment:
This license block can be declared as a constant for this module

```
O3DE_LICENSE_TEXT=\
"""'# {BEGIN_LICENSE}
# Copyright (c) Contributors to the Open 3D Engine Project.
# For complete copyright and license terms please see the LICENSE at the root of this distribution.
#
# SPDX-License-Identifier: Apache-2.0 OR MIT
# {END_LICENSE}
"""
```

and then

```
     d.write(O3DE_LICENSE_TEXT)
```


(This would of made my copyright scripts work better on the greps that it did)

Reply:
done